### DISCUSSION TITLE
Google Cloud Function Environment Variable

### DISCUSSION THREAD

**Discussion by josesuero** [quarkusio/quarkus#29964] [Q&A]
I'm using Quarkus to create a Google Cloud Functions and I'm following https://quarkus.io/guides/gcp-functions

it says if I have multiple functions I can put them all on the same project and set `quarkus.google-cloud-functions.function=functionName` on application.properties.

It also says: `This can be overridden with an environment variable too.` but it fails to say which environment variable it is, I have been surfing source code for a while with no luck.

Since I'm planning to deploy them with a Github Actions it's much easier to just change an environment variable that it would be the applications.properties

Can someone be as kind as to tell me which is the environment variable used to override the function. Also, would be nice if this is added to the guide.

Thanks In advance

Jose

[a comment by an automated account is not shown]

**Reply by loicmathieu** [reply]
Hi @josesuero any Quarkus config property can be overriden by environment variable.
The name of the variable is the name of the config property vonverted to be valid environment variable, see https://quarkus.io/guides/config-reference#environment-variables

Convertion rules is more or less capital letters and all special char converted to '_'.

So for `quarkus.google-cloud-functions.function` this should be `QUARKUS_GOOGLE_CLOUD_FUNCTIONS_FUNCTION`.

**Reply by josesuero** [reply]
Thanks!
